assets/app/views/directives/osc-key-values.html

```diff
@@ -34,38 +65,82 @@
-            <span class="key truncate">{{key}}</span>
-            <span ng-hide="editing">
-              <span class="value truncate">{{ value}}</span>
-              <a href="" ng-click="edit()" class="btn btn-default btn-xs" title="Edit">
+      
+      <div style="margin-top: 20px">
+        <!-- predefined keys -->
+        <div ng-repeat="(key,value) in entries | valuesIn:readonlyKeys">
+          <div layout row>
+            <div flex grow="5" shrink="5" class="truncate">{{key}}</div>
+            <div flex grow="5" shrink="5" class="truncate">{{value}}</div>
+            <div 
+              flex layout grow="1" shrink="1"
+              main-axis="end" cross-axis="baseline" 
+              style="flex-basis: 50px;">
+              &nbsp;  
+            </div>
+          </div>
+        </div>
+        
+        <!-- user defined keys -->
+        <div ng-repeat="(key,value) in entries | valuesNotIn:readonlyKeys">
+          <div layout row ng-controller="KeyValuesEntryController">
+            
+            <!-- view, edit/delete -->
+            <div flex grow="5" shrink="5" class="truncate">{{key}}</div>
+            
+            <div flex grow="5" shrink="5" class="truncate" ng-hide="editing">
+              {{value}}  
+            </div>
+
+            <div flex layout grow="1" shrink="1"
+              main-axis="end" cross-axis="baseline" ng-hide="editing" 
+              style="flex-basis: 50px;">
+              <a ng-click="edit()" 
+                 class="btn btn-default btn-xs" 
+                 title="Edit">
                 <i class="icon icon-pencil"></i>
               </a>
-              <a href="" ng-click="deleteEntry(key)" class="btn btn-default btn-xs" title="Delete" ng-if="allowDelete(key)">
+              <a ng-click="deleteEntry(key)" 
+                 class="btn btn-default btn-xs" 
+                 title="Delete" 
+                 ng-if="allowDelete(key)">
                 <i class="fa fa-times"></i>
               </a>
-            </span>
-            <span ng-show="editing" class="form-inline">
-              <div class="form-group">
-                  <input class="form-control" type="text" value="{{value}}" ng-model="value">
-                  <a href="" ng-click="update(key, value, $parent.entries)" class="btn btn-default" style="vertical-align: middle" title="Submit">
-                    <i class="icon icon-ok"></i>
-                  </a>
-                  <a href="" ng-click="cancel()" class="btn btn-default" style="vertical-align: middle" title="Cancel">
-                    <i class="icon icon-remove"></i>
-                  </a>
-              </div>
-            </span>
-          </span>
-        </li>
-      </ul>
+            </div>
+
+            <!-- editing view, save/cancel -->
+            <div layout row flex grow="5" shrink="5" ng-show="editing">
+              <input 
+                flex
+                type="text" 
+                value="{{value}}" 
```

Comment:
I'm going to break the js component out into a separate issue, per the conversation this morning (around it being one directive w/multiple controllers, rather than multiple directives).